extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/topage/ChunkDictionary.java

```diff
@@ -48,11 +50,12 @@ public interface Lookup<T> {
      */
     ChunkDictionary(
             @NotNull final Lookup<T> lookup,
-            @NotNull final Supplier<Dictionary> dictionarySupplier) {
+            @NotNull final Function<SeekableChannelsProvider.ChannelContext, Dictionary> dictionarySupplier) {
         this.lookup = lookup;
         this.dictionarySupplier = dictionarySupplier;
         this.valuesSupplier = new LazyCachingSupplier<>(() -> {
-            final Dictionary dictionary = dictionarySupplier.get();
+            // Dictionary is already materialized at this point, therefore, we can safely use NULL context
```

Comment:
This is getting more complex than I had originally thought, primarily because of `LongBitmapStringSet.ReversibleLookup` interface that this dictionary class implements, and then some other places where these methods are called. It feels weird adding a context argument in a map.
It would be easier for me explain this on call.

Reply:
Using a NULL context in ChunkDictionary now, as discussed offline.